src/workerd/api/streams/standard.c++

```diff
@@ -3070,20 +3146,29 @@ private:
                   return kj::mv(exception);
                 }).then(js, ioContext.addFunctor(
                     JSG_VISITABLE_LAMBDA(
-                      (readable=kj::mv(readable),pumpToReader=kj::mv(pumpToReader)),
+                      (readable=kj::mv(readable),
+                       pumpToReader=kj::mv(pumpToReader),
+                       accumulator=kj::mv(accumulator)),
                       (readable),
                       (jsg::Lock& js, kj::Maybe<jsg::Value> maybeException) mutable {
                   KJ_IF_SOME(reader, tryGetAs<PumpToReader>(pumpToReader)) {
                     auto& ioContext = reader.ioContext;
                     ioContext.requireCurrentOrThrowJs();
                     // Oh good, if we got here it means we're in the right IoContext and
                     // the PumpToReader is still alive.
-                    KJ_IF_SOME(exception, maybeException) {
-                      reader.doError(js, exception.getHandle(js));
+                    if (accumulator->pendingClose) {
+                      reader.doClose(js);
                     } else {
-                      // Else block to avert dangling else compiler warning.
+                      KJ_IF_SOME(exception, maybeException) {
+                        reader.doError(js, exception.getHandle(js));
+                      } else {
+                        // Else block to avert dangling else compiler warning.
```

Comment:
D'oh. I think adding KJ_SILENCE_DANGLING_ELSE_BEGIN / _END to JSG_VISITABLE_LAMBDA might fix this issue more generally. (It fixed it for me locally when I was puzzling over a similar case.)

Reply:
Will handle this separately